### PR TITLE
Jplana/k9s darwin xdg

### PR DESCRIPTION
**Pull request by jplana**
### Description

module k9s: Adds support for darwin configuration files without xdg (using the `~/Library/Application Support/k9s/` folder instead of `~/.config/k9s` in this platform)
module k9s: fix some typos in some of the configuration file names (I checked [latest published release](https://github.com/derailed/k9s/blob/v0.32.4/internal/config/files.go#L144) for this)

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

#### Maintainer CC

@katexochen
@liyangau 
@hm.maintainers.LucasWagler

### CODE REVIEW

**Comment by rycee** [comment]
Thanks for the contribution! Looks OK to me but I'll wait for a maintainer to have a look.

CC: @LucasWagler

File: modules/programs/k9s.nix [hunk 1]
```diff
@@ -162,33 +170,70 @@ in {
...
+        "k9s/skins/${name}.yaml"
+      else
+        "Library/Application Support/k9s/skins/${name}.yaml") {
+          source = yamlFormat.generate "k9s-skin-${name}.yaml" value;
+        }) cfg.skins;
+
   in mkIf cfg.enable {
     home.packages = [ cfg.package ];
 
     xdg.configFile = {
```
Comment (rycee) [inline modules/programs/k9s.nix:183]: Seems unnecessary to duplicate the condition for each file:
```suggestion
    xdg.configFile = mkIf (!(isDarwin && !config.xdg.enable)) {
```

Comment (rycee) [reply modules/programs/k9s.nix:183]: Or perhaps better, introduce a let binding

``` nix
enableXdgConfig = !isDarwin || config.xdg.enable
```

and have

```suggestion
    xdg.configFile = mkIf enableXdgConfig {
```

File: modules/programs/k9s.nix [hunk 2]
```diff
@@ -162,33 +170,70 @@ in {
...
+          source = yamlFormat.generate "k9s-plugin" cfg.plugin;
+        };
+
+      "k9s/views.yaml" =
+        mkIf (cfg.views != { } && !(isDarwin && !config.xdg.enable)) {
+          source = yamlFormat.generate "k9s-views" cfg.views;
+        };
+    } // (if !(isDarwin && !config.xdg.enable) then skinFiles else { });
+
+    home.file = {
```
Comment (rycee) [inline modules/programs/k9s.nix:211]:
``` nix
    home.file = mkIf (!enableXdgConfig) ({
      # …
    } // skinFiles)
```

File: modules/programs/k9s.nix (hunk 1 above)
Comment (jplana) [reply modules/programs/k9s.nix]: Hey! Thanks for taking the time to review the PR, and provide these suggestions :-) .  I added them as a new commit .

**Comment by LucasWagler** [comment]
I don't see tests for the new file paths. Looks good otherwise!

I'll sanity check it on my (non-Darwin) machine when I get the chance.

Out of the scope of this PR: Would it be worthwhile to create new Home Manager options like `darwin.configFile` and `darwin.configHome`, akin to the current `xdg.config*` options?

**Comment by jplana** [comment]
@LucasWagler regarding the tests for paths in darwin, I looked into it and I'm not sure If I'll be able to actually test the interesting logic. I assume that while running the tests I can do something along these lines:

```nix
 nmt.script = let
    configDir = if !pkgs.isDarwin then
      ".config/k9s"
        else
      "Library/Application Support/k9s";
    in
    ''
    assertFileExists "home-files/${configDir}/config.yaml"
...
...
```
however, I realized that the module running the tests is forcing `xdg.enable = true` (see [here](https://github.com/jplana/home-manager/blob/jplana%2Fk9s-darwin-xdg/tests/default.nix#L22)). I've even tried to override it with

```nix

let

  lib = import ../modules/lib/stdlib-extended.nix pkgs.lib;

in {
  programs.k9s.enable = true;
  xdg.enable = lib.mkForce false;
...
...
```

but when running the tests I keep getting

```
error: getting status of '/nix/store/1yky8nb7s3ia5i5qgasjfgpy0awy1r22-source/tests/modules/programs/modules': No such file or directory
```

**Comment by LucasWagler** [comment]
`lib` should be readily available without needing a manual `import` in the test file. Something like this:
```nix
{ config, lib, ... }:
{
  xdg.enable = lib.mkForce false;
  programs.k9s.enable = true;
...
```

**Comment by jplana** [comment]
Thanks @LucasWagler ! 

I added the tests for darwin paths without xdg enabled.

**Comment by LucasWagler** [comment]
Looks good on my machine! You may need to run the formatter one last time.

Thanks for your contribution!

**Review by LucasWagler — APPROVED** [review]

**Comment by jplana** [comment]
Formatting done! Thanks for your patience :-)

Finally, about creating the equivalent for `xdg.config*`, I looked into it. Probably it wouldn't make any sense for `xdg.configFile` as Apple specs is very clear to only allow "subdirectories" in `~/Library/Application Support` (see Apple's [Filesystem Programming Guide](https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/FileSystemOverview/FileSystemOverview.html#//apple_ref/doc/uid/TP40010672-CH2-SW1) for reference). As for `xdg.configHome`, `darwin.configHome`? `darwin.userAppSupport`? it might make sense if used as a constant so all references scattered through the home-manager codebase (and programs) use the same option. But I wouldn't expect it to change in the future, as even iOS has this very same FS organization and this folder seems to be backed up by the system there.

I you think this is worthwhile I can open a new PR and try my luck there.

**Comment by rycee** [comment]
Thanks! Merged to master now 🙂

**Pull request merged**